internal/api/courses/assignments/submissions/proxy/resubmit.go

```diff
@@ -0,0 +1,76 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/timestamp"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type ResubmitRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+	TargetSubmission string `json:"target-submission"`
+
+	ProxyUser core.TargetCourseUser `json:"proxy-email"`
+	ProxyTime *timestamp.Timestamp  `json:"proxy-time"`
+
+	Message string `json:"message"`
```

Comment:
I think we should probably just keep the message as is.
You could try to work out the semantics of dealing with this new message (i.e., the four possible cases) if you think it is important,
but I think it's easiest just to leave it off.